RiotSwiftUI/Modules/Room/PollHistory/View/PollHistory.swift

```diff
@@ -50,35 +46,91 @@ struct PollHistory: View {
         }
     }
     
+    @ViewBuilder
+    private var content: some View {
+        if viewModel.viewState.polls == nil {
+            loadingView
+        } else if viewModel.viewState.polls?.isEmpty == true {
+            noPollsView
+        } else {
+            pollListView
+        }
+    }
+    
     private var pollListView: some View {
         ScrollView {
             LazyVStack(spacing: 32) {
-                let enumeratedPolls = Array(viewModel.viewState.polls.enumerated())
-                
-                ForEach(enumeratedPolls, id: \.offset) { _, pollData in
+                ForEach(viewModel.viewState.polls ?? []) { pollData in
                     PollListItem(pollData: pollData)
                 }
                 .frame(maxWidth: .infinity, alignment: .leading)
-                
-                Button {
-                    #warning("handle action")
-                } label: {
-                    Text("Load more polls")
-                }
-                .frame(maxWidth: .infinity, alignment: .leading)
+
+                loadMoreButton
+                    .frame(maxWidth: .infinity, alignment: .leading)
             }
             .padding(.top, 32)
             .padding(.horizontal, 16)
         }
     }
     
+    private var loadMoreButton: some View {
+        HStack(spacing: 8) {
+            if viewModel.viewState.isLoading {
+                spinner
+            }
+            
+            Button {
+                #warning("handle action in next ticket")
+            } label: {
+                Text("Load more polls")
```

Comment:
I planned to to that in the next ticket dealing with "loading more".
Done it here because it's easy to forget! 😄
https://github.com/vector-im/element-ios/pull/7293/commits/6374ca137d4a850fbd85873c06f4e93d6a53b72a